### ISSUE TITLE
Getting Invalid username or password error 

### ISSUE THREAD

**Issue by 0xc0d3r** [semantic-release/semantic-release#1485]
## Current behavior

I installed `semantic-release-cli` globally and set up the `semantic-release` for our private GitHub repository.
```
semantic-release-cli setup
? What is your npm registry? https://npm.pkg.github.com/
? What is your npm username? xc0d3r
? What is your npm password? <MY-GITHUB-ACCESS-TOKEN>
? What is your GitHub username? 0xc0d3r
? What is your GitHub password? <MY-GITHUB-PASSWORD>
? What is your GitHub two-factor authentication code? [hidden]
? What CI are you using? Other
```
And I got the below secrets

```
GH_TOKEN=some-random-hash-value
NPM_TOKEN=some-random-hash-value
```

Next, I created a PR to our base branch `master` and I merged the PR. Now, `Release` workflow got triggered and we got this error 
```
An error occurred while running semantic-release: Error: Command failed with exit code 128: git ls-remote --heads https://[secure]@www.github.com/wingspanHQ/ws-logger
remote: Invalid username or password.
fatal: Authentication failed for 'https://github.com/wingspanHQ/ws-logger.git/
```

**NOTE:**  SAML is enabled to our GitHub organization. I authorized my personal access token as well.

Any specific configuration needed if we use SAML?

This is my `release.yml` file
```
name: Release
on:
  push:
    branches:
      - master
jobs:
  release:
    name: Release
    runs-on: ubuntu-18.04
    steps:
      - name: Checkout
        uses: actions/checkout@v1
      - name: Setup Node.js
        uses: actions/setup-node@v1
        with:
          node-version: 12
      - name: Install dependencies
        run: yarn install
      - name: Release
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          GH_TOKEN: ${{ secrets.GH_TOKEN }}
          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
        run: yarn semantic-release
```

## Environment

- **semantic-release** version: ^17.0.4
- CI environment: GitHub Actions
- Plugins used: No additional plugins
- CI logs: 
```
2020-03-10T11:31:50.0199526Z ##[group]Run yarn semantic-release --debug
2020-03-10T11:31:50.0199697Z [36;1myarn semantic-release --debug[0m
2020-03-10T11:31:50.0230734Z shell: /bin/bash -e {0}
2020-03-10T11:31:50.0230854Z env:
2020-03-10T11:31:50.0231549Z   GH_TOKEN: ***
2020-03-10T11:31:50.0231685Z   NPM_TOKEN: ***
2020-03-10T11:31:50.0232016Z   GITHUB_TOKEN: ***
2020-03-10T11:31:50.0232139Z ##[endgroup]
2020-03-10T11:31:50.1600645Z yarn run v1.22.0
2020-03-10T11:31:50.2063463Z $ semantic-release --debug
2020-03-10T11:31:50.4348258Z [11:31:50 AM] [semantic-release] â€º â„¹  Running semantic-release version 17.0.4
2020-03-10T11:31:50.4434048Z 2020-03-10T11:31:50.443Z semantic-release:config load config from: undefined
2020-03-10T11:31:50.4472840Z 2020-03-10T11:31:50.446Z semantic-release:config options values: {
2020-03-10T11:31:50.4473338Z   branches: [
2020-03-10T11:31:50.4473829Z     '+([0-9])?(.{+([0-9]),x}).x',
2020-03-10T11:31:50.4474258Z     'master',
2020-03-10T11:31:50.4475104Z     'next',
2020-03-10T11:31:50.4475603Z     'next-major',
2020-03-10T11:31:50.4486539Z     { name: 'beta', prerelease: true },
2020-03-10T11:31:50.4487141Z     { name: 'alpha', prerelease: true }
2020-03-10T11:31:50.4487458Z   ],
2020-03-10T11:31:50.4488024Z   repositoryUrl: 'https://www.github.com/wingspanHQ/ws-logger',
2020-03-10T11:31:50.4488527Z   tagFormat: 'v${version}',
2020-03-10T11:31:50.4488850Z   plugins: [
2020-03-10T11:31:50.4489345Z     '@semantic-release/commit-analyzer',
2020-03-10T11:31:50.4489849Z     '@semantic-release/release-notes-generator',
2020-03-10T11:31:50.4490321Z     '@semantic-release/npm',
2020-03-10T11:31:50.4490787Z     '@semantic-release/github'
2020-03-10T11:31:50.4491077Z   ],
2020-03-10T11:31:50.4491347Z   _: [],
2020-03-10T11:31:50.4491628Z   debug: true,
2020-03-10T11:31:50.4492107Z   '$0': 'node_modules/.bin/semantic-release'
2020-03-10T11:31:50.4492424Z }
2020-03-10T11:31:50.6176456Z 2020-03-10T11:31:50.617Z semantic-release:plugins options for @semantic-release/npm/verifyConditions: {}
2020-03-10T11:31:50.6182928Z [11:31:50 AM] [semantic-release] â€º âœ”  Loaded plugin "verifyConditions" from "@semantic-release/npm"
2020-03-10T11:31:50.6184397Z 2020-03-10T11:31:50.618Z semantic-release:plugins options for @semantic-release/github/verifyConditions: {}
2020-03-10T11:31:50.6186602Z [11:31:50 AM] [semantic-release] â€º âœ”  Loaded plugin "verifyConditions" from "@semantic-release/github"
2020-03-10T11:31:50.6188776Z 2020-03-10T11:31:50.618Z semantic-release:plugins options for @semantic-release/commit-analyzer/analyzeCommits: {}
2020-03-10T11:31:50.6189943Z [11:31:50 AM] [semantic-release] â€º âœ”  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
2020-03-10T11:31:50.6192054Z 2020-03-10T11:31:50.619Z semantic-release:plugins options for @semantic-release/release-notes-generator/generateNotes: {}
2020-03-10T11:31:50.6193726Z [11:31:50 AM] [semantic-release] â€º âœ”  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
2020-03-10T11:31:50.6194655Z 2020-03-10T11:31:50.619Z semantic-release:plugins options for @semantic-release/npm/prepare: {}
2020-03-10T11:31:50.6195835Z [11:31:50 AM] [semantic-release] â€º âœ”  Loaded plugin "prepare" from "@semantic-release/npm"
2020-03-10T11:31:50.6197053Z 2020-03-10T11:31:50.619Z semantic-release:plugins options for @semantic-release/npm/publish: {}
2020-03-10T11:31:50.6198188Z [11:31:50 AM] [semantic-release] â€º âœ”  Loaded plugin "publish" from "@semantic-release/npm"
2020-03-10T11:31:50.6199110Z 2020-03-10T11:31:50.619Z semantic-release:plugins options for @semantic-release/github/publish: {}
2020-03-10T11:31:50.6200141Z [11:31:50 AM] [semantic-release] â€º âœ”  Loaded plugin "publish" from "@semantic-release/github"
2020-03-10T11:31:50.6201276Z 2020-03-10T11:31:50.620Z semantic-release:plugins options for @semantic-release/npm/addChannel: {}
2020-03-10T11:31:50.6202681Z [11:31:50 AM] [semantic-release] â€º âœ”  Loaded plugin "addChannel" from "@semantic-release/npm"
2020-03-10T11:31:50.6203720Z 2020-03-10T11:31:50.620Z semantic-release:plugins options for @semantic-release/github/addChannel: {}
2020-03-10T11:31:50.6205439Z [11:31:50 AM] [semantic-release] â€º âœ”  Loaded plugin "addChannel" from "@semantic-release/github"
2020-03-10T11:31:50.6206200Z 2020-03-10T11:31:50.620Z semantic-release:plugins options for @semantic-release/github/success: {}
2020-03-10T11:31:50.6207047Z [11:31:50 AM] [semantic-release] â€º âœ”  Loaded plugin "success" from "@semantic-release/github"
2020-03-10T11:31:50.6208144Z 2020-03-10T11:31:50.620Z semantic-release:plugins options for @semantic-release/github/fail: {}
2020-03-10T11:31:50.6209254Z [11:31:50 AM] [semantic-release] â€º âœ”  Loaded plugin "fail" from "@semantic-release/github"
2020-03-10T11:31:50.8529165Z 2020-03-10T11:31:50.852Z semantic-release:git Error: Command failed with exit code 128: git push --dry-run --no-verify https://www.github.com/wingspanHQ/ws-logger HEAD:master
2020-03-10T11:31:50.8530203Z remote: Invalid username or password.
2020-03-10T11:31:50.8531207Z fatal: Authentication failed for 'https://github.com/wingspanHQ/ws-logger.git/'
2020-03-10T11:31:50.8532148Z     at makeError (/home/runner/work/ws-logger/ws-logger/node_modules/execa/lib/error.js:58:11)
2020-03-10T11:31:50.8532848Z     at handlePromise (/home/runner/work/ws-logger/ws-logger/node_modules/execa/index.js:114:26)
2020-03-10T11:31:50.8533315Z     at processTicksAndRejections (internal/process/task_queues.js:97:5)
2020-03-10T11:31:50.8534039Z     at async verifyAuth (/home/runner/work/ws-logger/ws-logger/node_modules/semantic-release/lib/git.js:207:5)
2020-03-10T11:31:50.8534798Z     at async module.exports (/home/runner/work/ws-logger/ws-logger/node_modules/semantic-release/lib/get-git-auth-url.js:43:5)
2020-03-10T11:31:50.8536454Z     at async run (/home/runner/work/ws-logger/ws-logger/node_modules/semantic-release/index.js:55:27)
2020-03-10T11:31:50.8537239Z     at async module.exports (/home/runner/work/ws-logger/ws-logger/node_modules/semantic-release/index.js:259:22)
2020-03-10T11:31:50.8537905Z     at async module.exports (/home/runner/work/ws-logger/ws-logger/node_modules/semantic-release/cli.js:55:5)
2020-03-10T11:31:51.0084218Z [11:31:51 AM] [semantic-release] â€º âœ–  An error occurred while running semantic-release: Error: Command failed with exit code 128: git ls-remote --heads https://[secure]@www.github.com/wingspanHQ/ws-logger
2020-03-10T11:31:51.0084647Z remote: Invalid username or password.
2020-03-10T11:31:51.0085171Z fatal: Authentication failed for 'https://github.com/wingspanHQ/ws-logger.git/'
2020-03-10T11:31:51.0085925Z     at makeError (/home/runner/work/ws-logger/ws-logger/node_modules/execa/lib/error.js:58:11)
2020-03-10T11:31:51.0086365Z     at handlePromise (/home/runner/work/ws-logger/ws-logger/node_modules/execa/index.js:114:26)
2020-03-10T11:31:51.0086601Z     at processTicksAndRejections (internal/process/task_queues.js:97:5)
2020-03-10T11:31:51.0087148Z     at async getBranches (/home/runner/work/ws-logger/ws-logger/node_modules/semantic-release/lib/git.js:66:11)
2020-03-10T11:31:51.0087605Z     at async module.exports (/home/runner/work/ws-logger/ws-logger/node_modules/semantic-release/lib/branches/expand.js:6:23)
2020-03-10T11:31:51.0088054Z     at async module.exports (/home/runner/work/ws-logger/ws-logger/node_modules/semantic-release/lib/branches/index.js:14:26)
2020-03-10T11:31:51.0088462Z     at async run (/home/runner/work/ws-logger/ws-logger/node_modules/semantic-release/index.js:56:22)
2020-03-10T11:31:51.0088899Z     at async module.exports (/home/runner/work/ws-logger/ws-logger/node_modules/semantic-release/index.js:259:22)
2020-03-10T11:31:51.0089323Z     at async module.exports (/home/runner/work/ws-logger/ws-logger/node_modules/semantic-release/cli.js:55:5) {
2020-03-10T11:31:51.0089790Z   shortMessage: 'Command failed with exit code 128: git ls-remote --heads https://[secure]@www.github.com/wingspanHQ/ws-logger',
2020-03-10T11:31:51.0090191Z   command: 'git ls-remote --heads https://[secure]@www.github.com/wingspanHQ/ws-logger',
2020-03-10T11:31:51.0090355Z   exitCode: 128,
2020-03-10T11:31:51.0090585Z   signal: undefined,
2020-03-10T11:31:51.0090741Z   signalDescription: undefined,
2020-03-10T11:31:51.0091001Z   stdout: '',
2020-03-10T11:31:51.0091305Z   stderr: 'remote: Invalid username or password.\n' +
2020-03-10T11:31:51.0091691Z     "fatal: Authentication failed for 'https://github.com/wingspanHQ/ws-logger.git/'",
2020-03-10T11:31:51.0091860Z   failed: true,
2020-03-10T11:31:51.0092001Z   timedOut: [secure],
2020-03-10T11:31:51.0092147Z   isCanceled: [secure],
2020-03-10T11:31:51.0092289Z   killed: [secure]
2020-03-10T11:31:51.0092428Z }
2020-03-10T11:31:51.0093444Z Error: Command failed with exit code 128: git ls-remote --heads https://[secure]@www.github.com/wingspanHQ/ws-logger
2020-03-10T11:31:51.0094042Z remote: Invalid username or password.
2020-03-10T11:31:51.0094701Z fatal: Authentication failed for 'https://github.com/wingspanHQ/ws-logger.git/'
2020-03-10T11:31:51.0095322Z     at makeError (/home/runner/work/ws-logger/ws-logger/node_modules/[4mexeca[24m/lib/error.js:58:11)
2020-03-10T11:31:51.0095994Z     at handlePromise (/home/runner/work/ws-logger/ws-logger/node_modules/[4mexeca[24m/index.js:114:26)
2020-03-10T11:31:51.0096661Z [90m    at processTicksAndRejections (internal/process/task_queues.js:97:5)[39m
2020-03-10T11:31:51.0097436Z     at async getBranches (/home/runner/work/ws-logger/ws-logger/node_modules/[4msemantic-release[24m/lib/git.js:66:11)
2020-03-10T11:31:51.0098227Z     at async module.exports (/home/runner/work/ws-logger/ws-logger/node_modules/[4msemantic-release[24m/lib/branches/expand.js:6:23)
2020-03-10T11:31:51.0098888Z     at async module.exports (/home/runner/work/ws-logger/ws-logger/node_modules/[4msemantic-release[24m/lib/branches/index.js:14:26)
2020-03-10T11:31:51.0099495Z     at async run (/home/runner/work/ws-logger/ws-logger/node_modules/[4msemantic-release[24m/index.js:56:22)
2020-03-10T11:31:51.0100117Z     at async module.exports (/home/runner/work/ws-logger/ws-logger/node_modules/[4msemantic-release[24m/index.js:259:22)
2020-03-10T11:31:51.0100743Z     at async module.exports (/home/runner/work/ws-logger/ws-logger/node_modules/[4msemantic-release[24m/cli.js:55:5) {
2020-03-10T11:31:51.0101424Z   shortMessage: [32m'Command failed with exit code 128: git ls-remote --heads https://[secure]@www.github.com/wingspanHQ/ws-logger'[39m,
2020-03-10T11:31:51.0102007Z   command: [32m'git ls-remote --heads https://[secure]@www.github.com/wingspanHQ/ws-logger'[39m,
2020-03-10T11:31:51.0102600Z   exitCode: [33m128[39m,
2020-03-10T11:31:51.0103033Z   signal: [90mundefined[39m,
2020-03-10T11:31:51.0103493Z   signalDescription: [90mundefined[39m,
2020-03-10T11:31:51.0103916Z   stdout: [32m''[39m,
2020-03-10T11:31:51.0104411Z   stderr: [32m'remote: Invalid username or password.\n'[39m +
2020-03-10T11:31:51.0104991Z     [32m"fatal: Authentication failed for 'https://github.com/wingspanHQ/ws-logger.git/'"[39m,
2020-03-10T11:31:51.0105451Z   failed: [33mtrue[39m,
2020-03-10T11:31:51.0105903Z   timedOut: [33m[secure][39m,
2020-03-10T11:31:51.0106330Z   isCanceled: [33m[secure][39m,
2020-03-10T11:31:51.0106778Z   killed: [33m[secure][39m
2020-03-10T11:31:51.0264873Z }error Command failed with exit code 1.
2020-03-10T11:31:51.0265871Z info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
2020-03-10T11:31:51.0347958Z ##[error]Process completed with exit code 1.
2020-03-10T11:31:51.0492100Z Post job cleanup.
2020-03-10T11:31:51.1443259Z [command]/usr/bin/git version
2020-03-10T11:31:51.1505962Z git version 2.25.1
2020-03-10T11:31:51.1531347Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2020-03-10T11:31:51.1559606Z http.https://github.com/.extraheader
2020-03-10T11:31:51.1569690Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2020-03-10T11:31:51.1643782Z Cleaning up orphan processes```

**Comment by 0xc0d3r** [comment]
I resolved the problem. In our `package.json` file, the `repository` field contains `www` and I removed it. 

```
{
...,
"repository": "https://github.com/wingspanHQ/ws-logger",
...
}
```

Don't how `www` added and don't know why `semantic-release` is throwing errors if there is `www`. Curious to know the reasons so that I can submit a postmortem report to my team.

**Comment by gr2m** [comment]
`semantic-release` is using a library to normalize the `"repository"` field in package.json, and I guess it doesn't catche the case when you use `www` in the repository URL. It's not something we'd try to catch in semantic-release specifically, but it would be interesting to know why the error you got was a 401 instead of an error explaining that the repository field is invalid.

**Comment by 0xc0d3r** [comment]
> but it would be interesting to know why the error you got was a 401 instead of an error explaining that the repository field is invalid

I too have the same question. Maybe in the process of generalizing the error messages, this case is mixed up in that library. Can you tell me which library is being used to normalize the `repository` field? I'd like to investigate this and submit a PR because I spent around ~4hrs to debug this and nobody should spend their time on this kind of bugs.

**Comment by gr2m** [comment]
I'm not 100% sure, but I think it's [https://github.com/sindresorhus/read-pkg-up](https://github.com/sindresorhus/read-pkg-up): 

https://github.com/semantic-release/semantic-release/blob/da5d1cf9c6e39291d5a352a9f6a7c200e1bbce5a/package.json#L45

We also use [`read-pkg`](https://github.com/sindresorhus/read-pkg) in `@semantic-release/npm`

https://github.com/semantic-release/npm/blob/6ffff197a4a8bd93d30ffa4963c96bdc761cfea2/package.json#L29

Let me know if that helps

**Comment by gr2m** [comment]
I just gave the underlying package a try, and it seems to correctly remove the `www` from the repository URL, see https://runkit.com/gr2m/semantic-release-1485

🤔

**Comment by gr2m** [comment]
could you share the full package.json file that you had when you saw the authorization problem?

**Comment by 0xc0d3r** [comment]
```
{
  "name": "@wingspanHQ/ws-logger",
  "version": "1.0.0",
  "description": "Wingspan's Logger",
  "main": "dist/index.js",
  "files": [
    "dist"
  ],
  "types": "dist/index.d.ts",
  "scripts": {
    "build": "tsc",
    "test": "ts-node test/write-console-log.ts",
    "lint": "tslint --project . -s node_modules/custom-tslint-formatters/formatters -t grouped 'src/**/*.+(ts|js)'",
    "prepublish": "yarn build",
    "semantic-release": "semantic-release"
  },
  "repository": "https://www.github.com/wingspanHQ/ws-logger",
  "author": "Gregory Maglio",
  "license": "MIT",
  "private": false,
  "devDependencies": {
    "@commitlint/cli": "^8.3.5",
    "@commitlint/config-conventional": "^8.3.4",
    "@types/mocha": "^7.0.2",
    "husky": "^4.2.3",
    "semantic-release": "^17.0.4",
    "typescript": "^3.8.3"
  },
  "husky": {
    "hooks": {
      "pre-receive": "yarn lint",
      "pre-commit": "yarn lint",
      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS"
    }
  },
  "engines": {
    "node": ">=10.0.0",
    "npm": "5.0.0"
  },
  "dependencies": {
    "@google-cloud/logging-bunyan": "^2.0.3",
    "bunyan": "^1.8.12",
    "custom-tslint-formatters": "^2.4.0",
    "debug": "^4.1.1",
    "tslint": "^6.0.0"
  },
  "publishConfig": {
    "registry": "https://npm.pkg.github.com/",
    "access": "restricted"
  }
}

```

**Comment by 0xc0d3r** [comment]
@gr2m One more error I got once I fixed the `www` issue.

```
npm ERR! code E400
npm ERR! 400 Bad Request - PUT https://npm.pkg.github.com/@wingspan%2fws-logger - scope 'wingspan' in package name '@wingspan/ws-logger' does not match repo owner 'wingspanHQ' in repository element in package.json
```

Later I changed the package.json `name` field and ensured that the scope and repo owner name is same.

**Comment by gr2m** [comment]
Ah I totally missed that you set it up for GitHub's registry, that might be related. I haven't tested the CLI setup with that myself yet.

I couldn't find a problem with the normalization of the `"registry"` field in package.json, it looks correct when I test it with your package.json content.

> Any specific configuration needed if we use SAML?

Is that a GitHub Enterprise Cloud feature? I have no experience with it myself.

I no longer believe that the `www` in the repository was the problem, I think it was sth else

**Comment by 0xc0d3r** [comment]
> Is that a GitHub Enterprise Cloud feature? I have no experience with it myself.

Yes. https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/enabling-and-testing-saml-single-sign-on-for-your-organization

I tried many solutions to solve the issue. As far as I remember, I did below steps

1. Removed the `www` in `repository` field
2. Run `semantic-release-cli setup` again
3. Update secrets in my GitHub repo
4. Keep a PR and merge
